lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -277,26 +292,29 @@ def get_direct_collection_fqcn(self, collection):
             # NOTE: should it be something like "<virtual>"?
             return None
 
-        return '.'.join((  # type: ignore[type-var]
-            self._get_direct_collection_namespace(collection),  # type: ignore[arg-type]
+        return '.'.join((
+            self._get_direct_collection_namespace(collection),
             self._get_direct_collection_name(collection),
         ))
 
-    def get_direct_collection_version(self, collection):
-        # type: (Collection) -> str
+    def get_direct_collection_version(self, collection: Collection) -> str:
         """Extract version from the given on-disk collection artifact."""
-        return self.get_direct_collection_meta(collection)['version']  # type: ignore[return-value]
+        return self.get_direct_collection_meta(collection)['version']
 
-    def get_direct_collection_dependencies(self, collection):
-        # type: (t.Union[Candidate, Requirement]) -> dict[str, str]
+    def get_direct_collection_dependencies(
+        self,
+        collection: Candidate | Requirement,
```

Comment:
Can this also use Collection, for consistency?

```suggestion
        collection: Collection,
```

Reply:
Maybe, I think in some places I just got the comments converted but in some places it was problematic. I'll check locally, thanks!